watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Tasks/TaskFragment.java

```diff
@@ -125,7 +125,7 @@ public void onRefresh() {
     @Override
     public void onListItemClick(ListView l, View v, int position, long id) {
         Task taskClicked;
-        if (TASK_TYPE == USER) taskClicked = mUserTaskList.get(position);
+        if (TASK_TYPE == USER && mUserTaskList.size() > 0) taskClicked = mUserTaskList.get(position);
```

Comment:
Shouldn't we switch on the id of the ListView, rather than checking whether or not the list exists?